### PR TITLE
[eclipse/xtext#1679]ported more code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679]ported more code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			}
+		}
+	}
+
+	public ITreeAppendable traceWithComments(ITreeAppendable appendable, EObject source) {
+		if (appendable instanceof TreeAppendable) {
+			return ((TreeAppendable) appendable).trace(source,
+					ILocationInFileProviderExtension.RegionDescription.INCLUDING_COMMENTS, false);
+		} else {
+			final ITextRegionWithLineInformation it;
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java:50]: ok

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/LoopParams.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		return this.separator = separator;
+	}
+
+	public Function1<? super ITreeAppendable, ? extends ITreeAppendable> setSuffix(
+			Function1<? super ITreeAppendable, ? extends ITreeAppendable> suffix) {
+		return this.suffix = suffix;
+	}
+
+	public Function1<? super ITreeAppendable, ? extends ITreeAppendable> setPrefix(String prefix) {
+		return this.prefix = (ITreeAppendable it) -> {
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/LoopParams.java:37]: A few unnecessary param types in the lambdas

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+public class TreeAppendableUtil {
+	@Inject
+	private ILocationInFileProvider locationProvider;
+
+	public ITreeAppendable traceSignificant(ITreeAppendable appendable, EObject source) {
+		return traceSignificant(appendable, source, false);
+	}
+
+	public ITreeAppendable traceSignificant(ITreeAppendable appendable, EObject source, boolean useForDebugging) {
+		if (appendable instanceof TreeAppendable) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java:30]: Looks like we could extract a single protected method here

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/imports/ImportsAcceptor.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.eclipse.xtend.lib.annotations.Accessors;
+import org.eclipse.xtext.common.types.JvmIdentifiableElement;
+import org.eclipse.xtext.common.types.JvmType;
+import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+
+/**
+ * @author dhuebner - Initial contribution and API
+ */
+public interface ImportsAcceptor {
+	@Accessors(AccessorType.PUBLIC_GETTER)
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/imports/ImportsAcceptor.java:24]: Unnecessary `@Accessors`

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/imports/StaticallyImportedMemberProvider.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+	public Iterable<JvmFeature> getAllFeatures(XImportDeclaration it) {
+		return getAllFeatures(it.eResource(), it.getImportedType(), it.isStatic(), it.isExtension(),
+				it.getMemberName());
+	}
+
+	public Iterable<JvmFeature> getAllFeatures(Resource resource, JvmDeclaredType importedType, boolean isStatic,
+			boolean extension, String memberName) {
+		if (!isStatic || importedType == null) {
+			return CollectionLiterals.<JvmFeature>emptyList();
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/imports/StaticallyImportedMemberProvider.java:58]: CollectionLiterals

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/jvmmodel/JvmIdentifiableMetaData.java [hunk 6]
```diff
@@ -20,22 +18,20 @@
...
-  
-  @Pure
-  public boolean isSynthetic() {
-    return this.synthetic;
-  }
-  
-  public void setSynthetic(final boolean synthetic) {
-    this.synthetic = synthetic;
-  }
+	@Accessors
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/jvmmodel/JvmIdentifiableMetaData.java:22]: Accessors

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/TreeAppendableUtil.java:30]: elses are not the same

**Pull request merged**